.github/workflows/release.yml

```diff
@@ -29,25 +32,146 @@ jobs:
+          echo OTACLIENT_WHL=${OTACLIENT_WHL} >> $GITHUB_OUTPUT
+          echo OTACLIENT_VERSION=${OTACLIENT_VERSION} >> $GITHUB_OUTPUT
+
       - name: build otaclient service API python package
         run: |
           pushd proto
           hatch build -t wheel
           popd
           cp proto/dist/*.whl dist
 
+      - name: upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-python-packages
+          path: dist/*.whl
+
+  build_squashfs_image_x86_64:
+    runs-on: ubuntu-latest
+    needs: build_python_packages
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: build squashfs image for x86_64
+        uses: ./.github/actions/build_squashfs_image
+        with:
+          platform: linux/amd64
+          platform_suffix: x86_64
+          base_image: ubuntu:22.04
+          whl: ${{ needs.build_python_packages.outputs.OTACLIENT_WHL }}
+          version: ${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}
+          output_squashfs: dist/ota-client-x86_64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+
+      - name: build patches
+        uses: ./.github/actions/build_patches
+        with:
+          platform_suffix: x86_64
+          version: ${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}
+          squashfs: dist/ota-client-x86_64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+          output_dir: dist
+
+      - name: upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-squashfs_x86_64
+          path: |
+            dist/*.squashfs
+            dist/*.patch
+
+  build_squashfs_image_arm64:
+    runs-on: ubuntu-latest
+    needs: build_python_packages
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: build squashfs image for arm64
+        uses: ./.github/actions/build_squashfs_image
+        with:
+          platform: linux/arm64
+          platform_suffix: arm64
+          base_image: arm64v8/ubuntu:22.04
+          whl: ${{ needs.build_python_packages.outputs.OTACLIENT_WHL }}
+          version: ${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}
+          output_squashfs: dist/ota-client-arm64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+
+      - name: build paches
+        uses: ./.github/actions/build_patches
+        with:
+          platform_suffix: arm64
+          version: ${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}
+          squashfs: dist/ota-client-arm64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+          output_dir: dist
+
+      - name: upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-squashfs_arm64
+          path: |
+            dist/*.squashfs
+            dist/*.patch
```

Comment:
Do you plan to use matrix to implement build for different architectures?
Something like:

```yaml
    strategy:
      fail-fast: false
      matrix:
        build_runner:
            - ubuntu:22.04
        architecture:
            - x86_64
            - arm64
      includes:
           - architecture: x86_64
             base_image: ubuntu:22.04
           - architecture: arm64
             base_image: arm64v8/ubuntu:22.04
```

Reply:
Good Idea, it simplifies yaml. Fixed.